### PR TITLE
ElastiCache documentation fixes and bringing some validations in line with documentation

### PR DESCRIPTION
**Pull request by gdavison**
Clarifies documentation around `aws_elasticache_cluster` forcing new resource when `node_type` is changed.

Also updates links to AWS documentation for the split between Redis and Memcached.

Additional cleanup.

Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
NONE
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSElasticacheReplicationGroup_\|TestAccAWSElasticacheCluster_'

--- PASS: TestAccAWSElasticacheReplicationGroup_clusteringAndCacheNodesCausesError (92.91s)
--- PASS: TestAccAWSElasticacheCluster_Engine_Memcached (736.74s)
--- PASS: TestAccAWSElasticacheReplicationGroup_ClusterMode_NonClusteredParameterGroup (797.02s)
--- PASS: TestAccAWSElasticacheCluster_multiAZInVpc (756.47s)
--- PASS: TestAccAWSElasticacheReplicationGroup_updateMaintenanceWindow (987.34s)
--- PASS: TestAccAWSElasticacheReplicationGroup_basic (990.40s)
--- PASS: TestAccAWSElasticacheReplicationGroup_updateParameterGroup (993.80s)
--- PASS: TestAccAWSElasticacheReplicationGroup_useCmkKmsKeyId (1275.23s)
--- PASS: TestAccAWSElasticacheReplicationGroup_vpc (1402.40s)
--- PASS: TestAccAWSElasticacheCluster_NumCacheNodes_Redis (13.79s)
--- PASS: TestAccAWSElasticacheReplicationGroup_ClusterMode_Basic (1544.07s)
--- PASS: TestAccAWSElasticacheReplicationGroup_Uppercase (874.66s)
--- PASS: TestAccAWSElasticacheReplicationGroup_enableAuthTokenTransitEncryption (1732.90s)
--- PASS: TestAccAWSElasticacheCluster_AZMode_Redis (692.47s)
--- PASS: TestAccAWSElasticacheReplicationGroup_multiAzInVpc (2078.44s)
--- PASS: TestAccAWSElasticacheCluster_AZMode_Memcached (676.45s)
--- PASS: TestAccAWSElasticacheReplicationGroup_tags (2185.35s)
--- PASS: TestAccAWSElasticacheReplicationGroup_NumberCacheClusters_Failover_AutoFailoverDisabled (2355.82s)
--- PASS: TestAccAWSElasticacheCluster_vpc (660.46s)
--- PASS: TestAccAWSElasticacheReplicationGroup_enableAtRestEncryption (2403.54s)
--- PASS: TestAccAWSElasticacheCluster_EngineVersion_Redis (1435.40s)
--- PASS: TestAccAWSElasticacheCluster_snapshotsWithUpdates (755.23s)
--- PASS: TestAccAWSElasticacheReplicationGroup_NumberCacheClusters_Failover_AutoFailoverEnabled (2488.60s)
--- PASS: TestAccAWSElasticacheReplicationGroup_redisClusterInVpc2 (2584.51s)
--- PASS: TestAccAWSElasticacheCluster_Port_Redis_Default (557.94s)
--- PASS: TestAccAWSElasticacheReplicationGroup_NumberCacheClusters (2658.04s)
--- PASS: TestAccAWSElasticacheCluster_ParameterGroupName_Default (599.39s)
--- PASS: TestAccAWSElasticacheReplicationGroup_updateDescription (1701.32s)
--- PASS: TestAccAWSElasticacheCluster_SecurityGroup_Ec2Classic (551.43s)
--- PASS: TestAccAWSElasticacheCluster_ReplicationGroupID_SingleReplica (1947.42s)
--- PASS: TestAccAWSElasticacheCluster_ReplicationGroupID_AvailabilityZone (1776.81s)
--- PASS: TestAccAWSElasticacheCluster_EngineVersion_Memcached (1511.04s)
--- PASS: TestAccAWSElasticacheCluster_NumCacheNodes_Decrease (929.59s)
--- PASS: TestAccAWSElasticacheCluster_ReplicationGroupID_MultipleReplica (2177.41s)
--- PASS: TestAccAWSElasticacheCluster_NodeTypeResize_Memcached (1375.49s)
--- PASS: TestAccAWSElasticacheCluster_Engine_Redis (663.86s)
--- PASS: TestAccAWSElasticacheCluster_Port (710.43s)
--- PASS: TestAccAWSElasticacheCluster_NumCacheNodes_IncreaseWithPreferredAvailabilityZones (1155.18s)
--- PASS: TestAccAWSElasticacheCluster_NumCacheNodes_Increase (1285.29s)
--- PASS: TestAccAWSElasticacheReplicationGroup_ClusterMode_NumNodeGroups (3796.91s)
--- PASS: TestAccAWSElasticacheReplicationGroup_updateNodeSize (3832.58s)
--- PASS: TestAccAWSElasticacheCluster_NodeTypeResize_Redis (1367.72s)
--- PASS: TestAccAWSElasticacheReplicationGroup_enableSnapshotting (898.99s)
```

### CODE REVIEW

File: aws/resource_aws_elasticache_cluster.go [hunk 1]
```diff
@@ -204,18 +209,18 @@ func resourceAwsElasticacheCluster() *schema.Resource {
...
 			},
-			// A single-element string list containing an Amazon Resource Name (ARN) that
-			// uniquely identifies a Redis RDB snapshot file stored in Amazon S3. The snapshot
-			// file will be used to populate the node group.
-			//
-			// See also:
-			// https://github.com/aws/aws-sdk-go/blob/4862a174f7fc92fb523fc39e68f00b87d91d2c3d/service/elasticache/api.go#L2079
 			"snapshot_arns": {
-				Type:     schema.TypeSet,
+				Type:     schema.TypeList,
```
Comment (bflad) [inline aws/resource_aws_elasticache_cluster.go:213]: Doesn't look like the logic change was applied for this type change:

```
	snaps := d.Get("snapshot_arns").(*schema.Set).List()
	if len(snaps) > 0 {
		s := expandStringList(snaps)
		req.SnapshotArns = s
		log.Printf("[DEBUG] Restoring Redis cluster from S3 snapshot: %#v", s)
	}
```

```suggestion
				Type:     schema.TypeSet,
```

**Review by bflad — CHANGES_REQUESTED** [review]
Otherwise, looks good. 👍

File: aws/resource_aws_elasticache_cluster.go (hunk 1 above)
Comment (gdavison) [reply aws/resource_aws_elasticache_cluster.go:213]: Yeah, missed changing the handling of the values. We don't have any tests for restoring from S3.

Created #17085

**Comment by bflad** [comment]
Ready for rebase

**Review by bflad — APPROVED** [review]
Looks good to me 🚀

**Pull request merged**